src/librbd/exclusive_lock/PreReleaseRequest.cc

```diff
@@ -100,7 +100,8 @@ void PreReleaseRequest<I>::send_set_require_lock() {
   // setting the lock as required will automatically cause the IO
   // queue to re-request the lock if any IO is queued
   if (m_image_ctx.clone_copy_on_read ||
-      m_image_ctx.test_features(RBD_FEATURE_JOURNALING)) {
+      m_image_ctx.test_features(RBD_FEATURE_JOURNALING) ||
+      m_image_ctx.test_features(RBD_FEATURE_DIRTY_CACHE)) {
```

Comment:
Is it possible for `cache::util::is_pwl_enabled()` to be true but `test_features(RBD_FEATURE_DIRTY_CACHE)` to be false here?  Just wondering if testing just the feature bit is enough.

Reply:
In init we check cache::util::is_pwl_enabled(). So i think if after init() we change pwl_enable from true to false, it don't effect later EXLOCK policy because we already used pwl_cache.  Is it ok?

Reply:
Yes, it is OK because the feature bit is authoritative.

I was wondering if the reverse is possible: outside of init, can the feature bit be not set but `pwl_enabled` be true?
I don't think it is possible but wanted to double check with you.

Reply:
I think it's impossible.